### ISSUE TITLE
canon-cups-ufr2 returns 404 when downloading driver

### ISSUE THREAD

**Issue by stefanDeveloper** [NixOS/nixpkgs#139559]
### Describe the bug
Currently, I cannot add the printer configuration to my nixos:

```
  services.printing.enable = true;
  services.printing.drivers = with pkgs; [ canon-cups-ufr2 ];
```

The `http://gdlp01.c-wss.com/gds/8/0100007658/08/linux-UFRII-drv-v370-uken-05.tar.gz` returns 404.  I guess this has to be updated here: https://github.com/NixOS/nixpkgs/blob/nixos-21.05/pkgs/misc/cups/drivers/canon/default.nix#L219 on Line 15.

### Steps To Reproduce
1. Add printer config with canon-cups-ufr2 to your `configuration.nix`
2. Rebuild nixos

### Expected behavior
Successful rebuild of nixos

### Screenshots
![image](https://user-images.githubusercontent.com/18898803/134820238-2773daa0-3379-4568-bf00-0012ddebe948.png)

### Additional context
```log
building '/nix/store/76409v6mwlcm6263yzjy6a5ivv94n7mi-linux-UFRII-drv-v370-uken-05.tar.gz.drv'...

trying http://gdlp01.c-wss.com/gds/8/0100007658/08/linux-UFRII-drv-v370-uken-05.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0    10    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download linux-UFRII-drv-v370-uken-05.tar.gz from any mirror
```

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/p5lnl4zr45n7mf9kz9w8yz3rqh001b5c-bash-interactive-4.4-p23-dev
copying path '/nix/store/p5lnl4zr45n7mf9kz9w8yz3rqh001b5c-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.67, NixOS, 21.05pre294818.2091b116423 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(stefan): `""`
 - channels(root): `"nixos-21.05.3339.056a3c1fae3, nixos-hardware, nixpkgs-21.05pre294818.2091b116423"`
 - nixpkgs: `/root/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: -
# a list of nixos modules affected by the problem
module: canon-cups-ufr2
```

**Comment by Artturin** [comment]
it was fixed in #135142 but wasn't backported
i now added the backport label so it will be backported

btw @kylesferrazza there's a update available but it changed the structure of the package therefore i will not update it
```diff
diff --git a/pkgs/misc/cups/drivers/canon/default.nix b/pkgs/misc/cups/drivers/canon/default.nix
index 93bc1f79e25..b6b5e031193 100644
--- a/pkgs/misc/cups/drivers/canon/default.nix
+++ b/pkgs/misc/cups/drivers/canon/default.nix
@@ -17,14 +17,22 @@ let
   i686_NIX_GCC = pkgsi686Linux.callPackage ({ gcc }: gcc) { };
   i686_libxml2 = pkgsi686Linux.callPackage ({ libxml2 }: libxml2) { };
 
-  commonVer = "4.10";
-  version = "3.70";
-  dl = "4/0100010264/01";
+  # to get the values for these
+  # https://asia.canon/en/support/0100924010
+  # ctrl+shift+i -> ctrl+shift+c -> select the download text and copy the data-url
+  # curl -I -S "THEURL" | grep location
+  # for example curl -I -S "https://pdisp01.c-wss.com/gdl/WWUFORedirectTarget.do?id=MDEwMDAwOTI0MDIw&cmp=ACB&lang=EN" | grep location
+
+  # save the final url with web.archive.org
+  # http://web.archive.org/web/20210928205219/http://gdlp01.c-wss.com/gds/0/0100009240/20/linux-UFRII-drv-v540-uken-21.tar.gz
+  # There is no more commonVer = "4.10";
+  version = "5.40";
+  dl = "0/0100009240/20";
 
   versionNoDots = builtins.replaceStrings [ "." ] [ "" ] version;
   src_canon = fetchurl {
-    url = "http://gdlp01.c-wss.com/gds/${dl}/linux-UFRII-drv-v${versionNoDots}-uken-07.tar.gz";
-    sha256 = "01nxpg3h1c64p5skxv904fg5c4sblmif486vkij2v62wwn6l65pz";
+    url = "http://gdlp01.c-wss.com/gds/${dl}/linux-UFRII-drv-v${versionNoDots}-uken-21.tar.gz";
+    sha256 = "sha256-P+Drf7hSLLoGUeWI0XIebVvxBs5DOdtRC3w8GL9YA+U=";
   };
 
 in
@@ -36,6 +44,7 @@ stdenv.mkDerivation {
   src = src_canon;
 
   postUnpack = ''
+    ls -R
     (cd $sourceRoot; tar -xzf Sources/cndrvcups-common-${commonVer}-1.tar.gz)
     (cd $sourceRoot; tar -xzf Sources/cndrvcups-lb-${version}-1.tar.gz)
   '';
```

**Comment by kylesferrazza** [comment]
> it was fixed in #135142 but wasn't backported i now added the backport label so it will be backported
> 
> btw @kylesferrazza there's a update available but it changed the structure of the package therefore i will not update it

Thanks for setting up the backport, @Artturin. I no longer have a canon printer for testing this so don't want to try to update to the new structure without being sure that I didn't break anything. Feel free to give it a go yourself (and replace me as maintainer) if you'd like.

**Comment by Artturin** [comment]
I don't have a printer 

If anyone wants to update it then this will be useful https://nixos.wiki/wiki/Nixpkgs/Reviewing_changes
